corehq/util/hmac_request.py

```diff
@@ -64,6 +66,7 @@ def _inner(request, *args, **kwargs):
             if expected_digest != digest:
                 return HttpResponse(status=401)
 
+            request.audit_user = audit_user
```

Comment:
Why not just do this before the `if ignore_if_debug` instead of doing it twice?

Reply:
Oh, because I only wanted to add this if the "system" user was authenticated correctly (or in the case of `ignore_if_debug` we're on an instance that's explicitly assuming all such requests are system requests without proof). If it returns a 401 response, we want it to be logged as `user` `null`.